ros2pkg/ros2pkg/build_type/__init__.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Wayne Parrott
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ros2cli.entry_points import load_entry_points
+from ros2cli.plugin_system import instantiate_extensions
+from ros2cli.plugin_system import PLUGIN_SYSTEM_VERSION
+from ros2cli.plugin_system import satisfies_version
+
+
+class BuildTypeExtension:
+    """
+    The interface for verb extensions.
+
+    The following properties must be defined:
+    * `NAME` (will be set to the entry point name)
+
+    The following methods must be defined:
+    * `main`
+    """
+
+    NAME = None
+    EXTENSION_POINT_VERSION = '0.1'
+    EXTENSION_POINT_NAME = 'ros2pkg.build_type'
+    NATIVE_BUILD_TYPES = ['cmake', 'ament_cmake', 'ament_python']
```

Comment:
I would suggest to not makes these build types special but implement them with an entry point too.

Reply:
Refactoring the existing build-types as extensions is a little bigger than I wanted to bite off in my first ros2 contribution. Let me propose that we move forward with this in 2 phases. Phase-1: intro the new build-type extension point - I already have a custom build-type that can demonstrate the value and serve as a guide for others.  Then in phase-2 we refactor the current 4 build-types to plugin via the build-type extension point? I will commit to helping impl phase-2 if this proposal is approved. Thoughts?